### PR TITLE
The user is now removed from 'LecturerInContest' and 'LecturerInContestCategory' as well.

### PR DESCRIPTION
**Pull request by deyordanov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1566

### CODE REVIEW

File: Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs [hunk 1]
```diff
@@ -88,15 +99,33 @@ public async Task<string> GetIdByName(string roleName)
...
+        var user = await this.usersDataService
+           .GetByIdQuery(model.UserId!)
+           .FirstAsync();
+
+        return (roleName, user);
+    }
+
+    private async Task RemoveUserFromLecturerInContestAndContestCategory(string roleName, UserProfile user)
+    {
+        if (roleName == Lecturer)
```
Comment (shsimeonova) [inline Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs:116]: Move this conditional in RolesBusinessService:85

File: Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs [hunk 2]
```diff
@@ -88,15 +99,33 @@ public async Task<string> GetIdByName(string roleName)
...
+                .GetQuery(licc => licc.LecturerId == user.Id)
+                .ToListAsync();
 
-         var user = await this.usersDataService
-            .GetByIdQuery(model.UserId!)
-            .FirstAsync();
+            this.lecturerInContestCategoryDataService.DeleteMany(categoriesUserIsLecturerIn);
 
-         return (roleName, user);
+            var contestsUserIsLecturerIn = await this.lecturerInContestsDataService
```
Comment (shsimeonova) [inline Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs:124]: This should be named userLecturerInContestsEntries or something like that..

**Review by shsimeonova — CHANGES_REQUESTED** [review]

File: Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs (hunk 1 above)
Comment (deyordanov) [reply Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs:116]: Done.

File: Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs (hunk 2 above)
Comment (deyordanov) [reply Services/Administration/OJS.Services.Administration.Business/Roles/RolesBusinessService.cs:124]: Done.

**Review by gogo4ds — APPROVED** [review]

**Review by shsimeonova — APPROVED** [review]

**Pull request merged**